src/coreclr/nativeaot/Runtime/unix/UnixContext.h

```diff
@@ -67,6 +67,22 @@ struct UNIX_CONTEXT
     uintptr_t GetIp() { return (uintptr_t)Pc(); }
     uintptr_t GetSp() { return (uintptr_t)Sp(); }
 
+    template <typename F>
+    void ForEachPossibleObjectRef(F lambda)
+    {
+        // it is doubtful anyone would implement X0-X28 not as a contiguous array
+        // just in case - here are some asserts.
+        ASSERT(&X0() + 1 == &X1());
+        ASSERT(&X0() + 10 == &X10());
+        ASSERT(&X0() + 20 == &X20());
+
+        for (uint64_t* pReg = &X0(); pReg <= &X28(); pReg++)
+            lambda((size_t*)pReg);
+
+        // Lr can be used as a scratch register
```

Comment:
What about FP?

Reply:
FP is supposed to form linked list by always [pointing to the next {fp, lr} pair](https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-160#stack-walking). I do not think we use it as a general purpose register.

Even the use of LR as a scratch is mostly theoretical. It is allowed, but it would have special behavior around calls.
@AndyAyersMS mentioned in the past that there is a possibility, so I am including it here just in case.

Reply:
> FP is supposed to form a linked list by always [pointing to the next {fp, lr} pair](https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-160#stack-walking).

That's Windows extension. I do not think it is required on Unix.

Reply:
I think use of FP as general purpose register would also mess up unwinding. 
It is easy to add it here though. 

It is also likely to be linearly stored next to the rest (since it is an alias for x29), so I could just loop through all X-es.

Reply:
> That's Windows extension. I do not think it is required on Unix.

Arm recommends the same pattern:
https://github.com/ARM-software/abi-aa/blob/main/aapcs64/aapcs64.rst#623the-frame-pointer

OS/compiler vendors are free to have their own ABIs, but I think JIT uses the same style regardless of OS here.

Reply:
Ok, you can keep it as is then.